lib/src/typer.dart

```diff
@@ -25,23 +31,32 @@ class TyperAnimatedText extends AnimatedText {
           duration: speed * text.characters.length,
         );
 
-  Animation<int> _typingText;
+  Animation<double> _typingText;
 
   @override
   Duration get remaining => speed * (textCharacters.length - _typingText.value);
 
   @override
   void initAnimation(AnimationController controller) {
-    _typingText = StepTween(
-      begin: 0,
-      end: textCharacters.length,
+    _typingText = CurveTween(
+      curve: curve,
     ).animate(controller);
   }
 
   /// Widget showing partial text, up to [count] characters
   @override
   Widget animatedBuilder(BuildContext context, Widget child) {
-    final count = _typingText.value;
+    /// Output of CurveTween is in the range [0, 1] for majority of the curves.
+    /// It is converted to [0, textCharacters.length].
+    var count = (_typingText.value * textCharacters.length).round();
+
+    if (count < 0) {
+      count = 0;
+    }
+    if (count >= textCharacters.length) {
```

Comment:
Could `count` ever exceed the `length`?  I don't think so.  As a result, I'd remove this.  (If I am wrong, then please add a comment to explain.)

Reply:
I guess the above example is applicable here as well. This statement clips the `curve` if it ever exceeds `length`.

Reply:
Got it.

Reply:
If you set the limit then it won't exceed the `length`